web/src/stream_create.ts

```diff
@@ -663,42 +667,39 @@ export function set_up_handlers(): void {
     folder_widget = stream_settings_components.set_up_folder_dropdown_widget();
 }
 
-export function initialize(): void {
-    $("#channels_overlay_container").on("click", "#archived_stream_rename", (e) => {
-        e.preventDefault();
-        e.stopPropagation();
-        const stream_id = Number.parseInt($("#archived_stream_rename").attr("data-stream-id")!, 10);
-        const stream = stream_data.get_sub_by_id(stream_id);
-
-        assert(stream !== undefined);
+export function open_edit_modal_for_archived_channel(stream_id: number): void {
```

Comment:
This code is still duplicated with the code for editing the info of a stream in general.

You can extract the code to open modal and save info for a stream in general in functions in a separate preparatory commit and then directly use those functions in this commit. `stream_edit.ts` would be a better place for those functions because they are about editing info of existing streams, but if there are some concerns around import cycles then they can be defined in common files like `stream_settings_components.ts`.